### PR TITLE
Update caching docs

### PR DESCRIPTION
**Pull request by priitlatt**
Sync caching docs with upcoming changes.

### CODE REVIEW

File: content/knowledge-codemagic/caching.md [hunk 1]
```diff
@@ -7,7 +7,9 @@ aliases:
   - /flutter-configuration/dependency-caching
 ---
 
-You can speed up your builds by storing dependencies on Codemagic. 
+Caching can improve the efficiency of your build and deployment processes on Codemagic by reusing artifacts, such as packages, libraries, and compiled code.
```
Comment (helinanever) [inline content/knowledge-codemagic/caching.md:10]: When I read "artifacts" here, I immediately think of build artifacts, but all the alternatives I can think of, such as "components" or "dependencies", don't work better either.

File: content/knowledge-codemagic/caching.md [hunk 2]
```diff
@@ -7,7 +7,9 @@ aliases:
...
+
+The caching process begins when a successful build generates a cache. Codemagic stores this cache for a maximum of 14 days. After 14 days, the cache expires and is no longer accessible for subsequent builds. When this occurs, your workflow will revert to building without cache, which means dependencies and artifacts need to be fetched and generated again. If a build successfully completes without cache, Codemagic will automatically generate a new cache based on the build's output. This new cache is then uploaded and can be used for the next 14 days, following the same process.
```
Comment (helinanever) [inline content/knowledge-codemagic/caching.md:12]: Your version is fine too but I felt some useful stuff could be moved earlier in the paragraph.

```suggestion
When you have enabled caching for a workflow, Codemagic will automatically generate a cache based on the output of the first successful build. This cache is stored for a maximum of 14 days. After 14 days, the cache expires and is no longer accessible for subsequent builds. When this occurs, your workflow will fetch all dependencies and artifacts again and generate a new cache. This new cache is then uploaded and can be used for the next 14 days, following the same process.
```

File: content/knowledge-codemagic/caching.md [hunk 3]
```diff
@@ -54,7 +56,11 @@ To use caching, simply add a `cache` section to your `codemagic.yaml` file and l
 
 ## Cache usage limits
 
-Caching is limited to **3GB** to avoid potential performance issues. Installing dependencies without using caching could be faster than retrieving or updating cached data.
+Maximum cache size is limited to
+- **10GB** per workflow for team account, and
```
Comment (helinanever) [inline content/knowledge-codemagic/caching.md:60]:
```suggestion
- **10GB** per workflow for teams, and 
```

**Review by helinanever — COMMENTED** [review]
Some nits

File: content/knowledge-codemagic/caching.md (hunk 1 above)
Comment (priitlatt) [reply content/knowledge-codemagic/caching.md:10]: What if "artifacts" is indeed replaced with "components", but we elaborate a little what is meant by that. Such as
> components that are generated or fetched during the build, such as packages, ...

Comment (helinanever) [reply content/knowledge-codemagic/caching.md:10]: If that makes sense to you, go ahead.

Comment (priitlatt) [reply content/knowledge-codemagic/caching.md:10]:
```suggestion
Caching can improve the efficiency of your build and deployment processes on Codemagic by reusing components that are generated or fetched during the build, such as packages, libraries, and compiled code.
```

**Review by helinanever — APPROVED** [review]
LGTM

**Pull request merged**